yrs/src/block_store.rs

```diff
@@ -398,25 +451,39 @@ impl BlockStore {
 
     /// Returns mutable reference to an item, given its pointer. Returns `None` if not such block
     /// could be found.
-    pub(crate) fn get_item_mut(&mut self, ptr: &block::BlockPtr) -> Option<&mut block::Item> {
-        let blocks = self.clients.get_mut(&ptr.id.client)?;
-        let block = blocks.list.get_mut(ptr.pivot())?;
+    pub(crate) fn get_item_mut(&self, ptr: &block::BlockPtr) -> Option<&mut block::Item> {
+        let block = self.get_block_mut(ptr)?;
```

Comment:
Previous implementation was constrained to use only `BlockPtr::pivot` field - pivots are not to be trusted and sometimes we may need to fallback using standard search.